### ISSUE TITLE
Improve Prediction Speed 

### ISSUE THREAD

**Issue by kshitijrajsharma** [hotosm/fAIr#107]
Currently it is taking approx 10sec which is good, But we want to make it best , so that predictions will be done in 3-5 sec 
Where should you be looking at to improve this ? 
- You should be looking here : https://github.com/hotosm/fAIr-utilities/tree/master/hot_fair_utilities
- fAIr utilities have several functions , it has post processing pre processing and running main training code ,Currently postprocessing speed is really fast , Loading model using keras.load_model is taking time , adopting h5 format has improved performance from 15-20 sec to 5-6 sec which is yet not very best , Prediction itself is taking time approx 5-6 sec , due to which total time is approx 10 11 sec , I think we can further optimize this may be keep model in memory for some time if there is continuous use case , we can limit no of models allowed to be in memory with respect to system specs , its just an idea may be there is better approach .

https://github.com/hotosm/fAIr-utilities/blob/a141c15f0255a9a30215123fa6912870600c0b79/hot_fair_utilities/inference/predict.py#L46

**Comment by mohwald** [comment]
Some of my ideas, from the simplest:
- Updating python and tensorflow to higher version can help. In my own experience, updating from pytorch 1.7 to 1.9 and python3.8 to 3.10 increased the model output speed about 10-20%, which can give 1 second.
- Checking, if server GPU runs on 100% and the memory usage is constant (`gpustat` or `watch -n 1 nvidia-smi`). If not, then data preprocessing can be improved to feed the data into the model. I have no clue how to check this on EC2.
- Use better, state-of-the-art model, instead of ramp. Currently, ViTs are the most popular, but they may perform worse with smaller model sizes. For fast prediction, YOLOs are popular. I will try to find and compare them. Another challenge here might be to find base model tailored to satellite/drone image segementation or to pretrain them.

**Comment by kshitijrajsharma** [comment]
@8ToThePowerOfMol , Upgrading python and tensorflow is good idea , Currently RAMP has list of numerous dependencies I wonder if we run in to dependencies tree conflict problem if we upgrade , sometimes they provide serious issue . Would be interesting to look into it 
-About GPU usage , right now GPU do consume 100 % of its memory when i do nvidia-smi , i guess we are fine from that end 
- Third point seems promising , It will be interesting to find out the comparison results  & accuracy ! and if they do perform better and we can run predictions within 4 5 sec that would be interesting

**Comment by omranlm** [comment]
Since we added the predictor repository @ https://github.com/hotosm/fairpredictor and we have now support for prediction using only CPU for RAMP at least .. This issue is not valid any more. 

For more about prediction check https://github.com/hotosm/fairpredictor
